python/ray/data/datasource/parquet_datasource.py

```diff
@@ -37,11 +39,34 @@
 PARQUET_READER_ROW_BATCH_SIZE = 100000
 FILE_READING_RETRY = 8
 
-# The estimated bytes size multiplier for reading Parquet data source in Arrow,
-# as Arrow in-memory representation uses much more memory compared to Parquet
-# uncompressed representation. See https://github.com/ray-project/ray/pull/26516
-# for more context.
-PARQUET_TO_ARROW_SIZE_MULTIPLIER = 5
+# The default size multiplier for reading Parquet data source in Arrow.
+# Parquet in-memory data format is encoded with various encoding techniques (such as
+# dictionary, RLE, delta), so Arrow in-memory representation uses much more memory
+# compared to Parquet in-memory representation. Parquet file statistics only record
```

Comment:
We are estimating the multiplier/ratio on top of Parquet's uncompressed/encoded data size, not Parquet compressed (on-disk) data size. So I think the wording currently should be right.

Reply:
Let me make sure we've got this right; we're talking about three possible compression states for Parquet data:
1. **Compressed + Parquet-encoded, on disk:** In addition to Parquet's native column encoding (dictionary, RLE, delta), the file is also compressed with e.g. Snappy or gzip.
2. **Parquet-encoded, on disk:** Just Parquet's native column encoding (dictionary, RLE, delta), no other compression.
3. **Parquet-encoded, in memory:** Just Parquet's native column encoding (dictionary, RLE, delta), no other compression.
4. **Arrow format, in memory:** Fully decompressed, including both Parquet's column encoding and any other compression, represented in Arrow's in-memory format.

It's my understanding that:
- Arrow is giving us (2),
- (2) and (3) are equivalent in terms of size in bytes,
- we don't read the full Parquet-encoded file into memory and then decode, we instead read in chunks of row groups that progressively decoded.

For those reasons, I was thinking that referring to this multiplier/ratio as being on top of "Parquet's on-disk representation" is more accurate, but I might be misunderstanding some aspect of this.

Reply:
`estimated-arrow-in-memory-size` = `multipler-in-this-PR` * `parquet-encoded-size`.
I agree with terminology used for (1) to (4). In practice, people always compress data when writing Parquet file, so there's really no (2): Parquet encode on-disk. When talking about Parquet in-memory size, it always means Parquet encoded data. 

I can change the phrase to `Parquet's encoded representation` if that sounds better.

Reply:
Ah I see where you were coming from with this language now, got it. I'd personally lean towards `Parquet's encoded representation` since it being on-disk or in-memory shouldn't matter, the multiplier is applied to a storage-agnostic "number of bytes for Parquet-encoded data", but feel free to leave it as is since that should be implied with "Parquet in-memory representation"!

Reply:
I also think "Parquet in-memory representation" is a bit inaccurate and can be read as just moving Parquet encoded data into memory bytes-by-bytes, or a new format for Parquet when it's in-memory. I think is a ratio of Arrow format / Parquet format, independent of storage medium.

Reply:
Let me update to `Parquet's encoded representation` to avoid confusion. Thanks all.

Reply:
updated.